test/test_dims_create.jl

```diff
@@ -0,0 +1,24 @@
+using Test
+using MPI
+
+MPI.Init()
+comm = MPI.COMM_WORLD
+nnodes = MPI.Comm_size(comm)
+ndims = 3
+
+cdims = Cint[0,0,0]
+MPI.Dims_create!(nnodes, ndims, cdims)
+@test prod(cdims) == nnodes
+
+cdims = Cint[1,0,1]
+MPI.Dims_create!(nnodes, ndims, cdims)
+@test cdims == Cint[1,nnodes,1]
+
+for dims in ([0,0,0], [0 0 0])
+    dims_t = typeof(dims)
+    MPI.Dims_create!(nnodes, dims)
+    @test typeof(dims) == dims_t  #Ensure the type of dims is as before
```

Comment:
This test is not needed.  Julia can't change the type of the argument in the caller scope (it can make a new variable within the function that has a different type, but the variable `dims` remains unaffected).

Reply:
OK, removed